src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -75,10 +75,11 @@ public void Initialize(StorageEnvironmentOptions options)
 
         private void InitializeInternal()
         {
-            IndexStore.Initialize();
+            var indexStoreTask = IndexStore.InitializeAsync();
             SqlReplicationLoader.Initialize();
             DocumentTombstoneCleaner.Initialize();
-            IndexStore.OpenIndexesTask.Wait(DatabaseShutdown);
+
+            indexStoreTask.Wait(DatabaseShutdown);
```

Comment:
Hm... if there is a failure in the init process, we need to make sure that we are properly disposing all resources.
That means that we need can't just dispose the IndexStore, we need to the let its InitializeAsync to run, then dispose it.